app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java

```diff
@@ -124,173 +108,111 @@ protected void onCreate(Bundle savedInstanceState) {
         });
-
-
-        // TODO: Move to fragment but then idk how to update the VibeEvent via imageIsSelected flag
-        // ---Remove Image Button---
-        removeImageBtn.setOnClickListener((View v) -> {
-                    imageFragment.setImageBitmap(null);
-                    imageIsSelected = false;
-        });
         
         //---Remove Location---
-        deleteLocationButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                vibeEvent.setLatitude(null);
-                vibeEvent.setLongitude(null);
-                mapFragment.clearMap();
-                mapFragment.hideMap();
-                deleteLocationButton.setVisibility(View.GONE);
-            }
-
+        deleteLocationButton.setOnClickListener((View v) -> {
+            clearLocation();
         });
 
         // --- Show Output on button click ---
         addBtn.setOnClickListener(view -> {
-
-            if (reasonField.getText().toString().trim().split("\\s").length <= REASON_FIELD_MAX_WORD_COUNT) {
-                reasonField.setError(null);
-                vibeEvent.setReason(reasonField.getText().toString());
-            } else {
-                reasonField.setError(String.format(Locale.CANADA, "Max %d words allowed", REASON_FIELD_MAX_WORD_COUNT));
-                return;
+            if (fieldAreValid()) {
+                addEditController.saveVibeEvent(vibe, reason, socSit, image, latitude, longitude);
+                finish();
             }
-
-            if (vibeEvent.getVibe() == null) {
-                Toast.makeText(getApplicationContext(), "Select a Vibe!", Toast.LENGTH_LONG).show();
-                return;
-            }
-
-
-            uploadVibeEvent();
-
-            finish();
         });
 
     }
 
-    @Override
-    public void onSocSitSelected(SocSit socSit) {
-        vibeEvent.setSocSit(socSit);
+    // ------------- Validation -------------
+    public boolean fieldAreValid() {
+        return vibeFieldIsValid() && reasonFieldIsValid();
     }
 
-    @Override
-    public void onImageSelected(Bitmap selectedImageBitmap) {
-        imageBitmap = selectedImageBitmap;
-        imageIsSelected = true;
+    public boolean vibeFieldIsValid() {
+        if (vibe == null) {
+            Toast.makeText(getApplicationContext(), "Select a Vibe!", Toast.LENGTH_LONG).show();
+            return false;
+        }
+
+        return true;
     }
 
-    @Override
-    public void onVibeSelected(Vibe vibe) {
-        vibeEvent.setVibe(vibe.getName());
-        setTheme(vibe);
+    public boolean reasonFieldIsValid() {
+        reason = reasonField.getText().toString().trim();
+        if (reason.split("\\s").length <= REASON_FIELD_MAX_WORD_COUNT) {
+            reasonField.setError(null);
+            return true;
+
+        } else {
+            reasonField.setError(String.format(Locale.CANADA, "Max %d words allowed", REASON_FIELD_MAX_WORD_COUNT));
```

Comment:
Did we want to make this a Toast message instead to be consistent with the rest of the code or are we cool with this?